refactor: improve poetry packaging

Scrumplex opened the pull request:
This allows poetry-only builds (i.e. `poetry install`) and better packaging for distributions.

Additionally this PR adds Nix packaging to allow users of Nix to build and run the app locally.

Scrumplex committed:
refactor: improve poetry packaging

Signed-off-by: Sefa Eyeoglu <contact@scrumplex.net>

Scrumplex committed:
chore: update ruff to 0.6.1

This version is compatible with poetry2nix while being just a few patch
versions behind.

Signed-off-by: Sefa Eyeoglu <contact@scrumplex.net>

Scrumplex committed:
feat(nix): add Nix flake

Signed-off-by: Sefa Eyeoglu <contact@scrumplex.net>

ShyAssassin commented:
this looks good to me!
could you please rebase this PR to the [features/algorithms](https://github.com/EyeTrackVR/ETVR-Backend/tree/feature/algorithms) branch as this is currently the active dev branch, i will do a full code review + testing later today!

Scrumplex commented:
I will be able to reduce the flake.nix once https://github.com/nix-community/poetry2nix/pull/1803 is merged

Scrumplex committed:
fix(camera): allow and resolve /dev/serial ports

This allows deterministic serial ports on Linux.

Users can define serial ports like
`/dev/serial/by-id/usb-Espressif_USB_JTAG_serial_debug_unit_<MAC address of ESP>-if00`
which will stay constant indefinitely. It also allows differentiating
different trackers based on their MAC address instead of relying on
`ttyACM0`, `tyyACM1`, ...

Signed-off-by: Sefa Eyeoglu <contact@scrumplex.net>

ShyAssassin reviewed (changes requested):
Everything looks good, the only thing i can comment on is that the `makefile` and `build.py` script have not been updated to reflect the backends new directory.

ShyAssassin commented on eyetrackvr_backend/main.py:
```diff
@@ -4,9 +4,9 @@
 from fastapi.staticfiles import StaticFiles
-from app.logger import setup_logger
+from .app.logger import setup_logger
 from fastapi import FastAPI
-from app.etvr import ETVR
+from .app.etvr import ETVR
```
making this import relative seems to break things with import errors and prevent startup when running from main directly

Scrumplex replied:
Hmm, how are you starting the application? I am running `python -m eyetrackvr_backend.main` which will treat the folder main resides in as a package

ShyAssassin replied:
i am running it directly with `poetry run python eyetrackvr_backend/main.py` but this relative package also cannot be found after being packaged with pyinstaller

Scrumplex replied:
I see. Let me see how we can bend pyinstaller to run it as a module instead (i.e. equivalent to `poetry run python -m eyetrackvr_backend.main`)

Scrumplex replied:
Seems like pyinstaller does not support modules properly? A workaround is to have a script outside the module directory that calls main.

Scrumplex commented:
Okay I changed the structure again. In my local testing, the pyinstaller binary works as well as running it manually (i.e. `poetry run eyetrackvr-backend`)

ShyAssassin commented:
Juts did a full test with my ETVR hardware locally and everything looks good!
there seems to be a bug related to the camera process not detecting a new source address when config is changed but i do not believe this PR introduced that bug.
